src/parse/generics.rs

```diff
@@ -457,9 +458,12 @@ impl SimpleGeneric {
 /// a const generic parameter, e.g. `struct Foo<const N: usize> { .. }`
 #[derive(Debug, Clone)]
 pub struct ConstGeneric {
-    const_token: Ident,
-    ident: Ident,
-    constraints: Vec<TokenTree>,
+    /// The `const` token for this generic
+    pub const_token: Ident,
+    /// The ident of this generic
+    pub ident: Ident,
+    /// The "constraints" (type) of this generic, e.g. `const N: usize`
```

Comment:
This would be `: usize`, as `const` and `N` would be stored in `const_token` and `ident` respectively

Reply:
Technically I think it's just `usize`, since the `:` gets eaten by `assume_punct(input.next(), ':');`. I couldn't figure out the best way to express that. Maybe:

```
/// The "constraints" (type) of this generic, e.g. the `usize` from `const N: usize`

Reply:
Looks good to me